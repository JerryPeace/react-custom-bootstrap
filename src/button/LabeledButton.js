import PropTypes from 'prop-types';
import React from 'react';
import Button from './Button';

const LabeledButton = (props) => {
    const {
        btnLabel,
        buttonText,
        children,
        ...rest
    } = props;

    return (
        <Button {...rest} >
            <span className='btn-label'>
                {btnLabel}
            </span>
            {buttonText}
            {children}
        </Button>
    );
};

LabeledButton.propTypes = {
    btnLabel: PropTypes.node,
    buttonText: PropTypes.node
};

export default LabeledButton;
