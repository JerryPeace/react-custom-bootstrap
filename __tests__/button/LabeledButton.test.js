import React from 'react';
import {mount} from 'enzyme';
import LabeledButton from 'button/LabeledButton';

describe('LabeledButton', () => {
    test('without children or buttonText', () => {
        const labeledbutton = mount(
            <LabeledButton btnLabel='Test Label' >
                children
            </LabeledButton>
        );
        expect(labeledbutton).toMatchSnapshot();
    });
});
