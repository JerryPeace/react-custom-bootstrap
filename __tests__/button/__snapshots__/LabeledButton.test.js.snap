// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledButton without children or buttonText 1`] = `
<LabeledButton
  btnLabel="Test Label"
>
  <Button
    active={false}
    block={false}
    bsClass="button"
    bsStyle="default"
    disabled={false}
    faProps={Object {}}
    isCaretStyle={false}
    isDropdown={false}
    isLoading={false}
    standalone={true}
    type="button"
  >
    <button
      className="btn btn-default"
      disabled={false}
      type="button"
    >
      <span
        className="btn-content"
      >
        <span
          className="btn-label"
        >
          Test Label
        </span>
        children
      </span>
    </button>
  </Button>
</LabeledButton>
`;
